GUI for login/booking

ccho-0508 opened the pull request:
bookRoom.py and test.kv
- bookRoom.py is for the executable file/build but can be run separately to create the login/booking GUI 
- test.kv is a separate file for styling 
- Code is kind of jank - **fix later**

ccho-0508 commented:
EXE file is called 'runBookingApp' in the exe folder

bobby-palko commented on exe/bookroom.spec:
```diff
@@ -0,0 +1,44 @@
+block_cipher = None
+
+
+a = Analysis(['/Users/ccho/Developer/libraryroomscheduler/proofOC/bookRoom.py'],
```
Having these as absolute file paths causes the build to fail, since my local copy of the project doesn't live in `/Users/ccho/Developer`

bobby-palko commented on exe/bookroom.spec:
```diff
@@ -0,0 +1,44 @@
+pyz = PYZ(a.pure, a.zipped_data,
+             cipher=block_cipher)
+
+exe = EXE(pyz, Tree('/Users/ccho/Developer/libraryroomscheduler/proofOC'),
```
Same thing here regarding the file path

bobby-palko commented on proofOC/bookRoom.py:
```diff
@@ -0,0 +1,174 @@
+
+        # Make Another Booking - Element xpath ://a[@href='/reserve/charles-small'] [@class='btn btn-primary']
+        
+        yourMomButton = WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, "//button[@id='s-lc-eq-bform-submit']")))
```
lol nice

bobby-palko commented on README.md:
````diff
@@ -1,28 +1,51 @@
-- What is expected to happen when the app start. 
+- Use this github repository
+- Use Visual Studio Code
+- Install:
+
+```
+pip3 install selenium
+pip3 install kivy
+pip3 install webdriver_manager
+pip3 install pyinstaller
+
+```
````
we may want to think about packaging some sort of virtual environment with these (and python) locked to a specific version. I'll start checking that out

bobby-palko reviewed (changes requested):
I'm not able to build the project locally since the file paths are absolute to your machine. Additionally, testing `bookRoom.py` also fails; I'm getting a "Login Error" (it looks like the try/except on L 147 is failing for some reason).

One last thing, can you rebase this onto master before we merge? Otherwise it'll wipe Brendan's and my personas and the `.gitignore`.

ccho-0508 commented:
Ok I changed the paths and updated the build instructions! I forgot to say but I put the personas in a separate folder so your work is still there. It's called 'personas' and the .gitignore includes it.

Also I'm not sure why the login is failing but it may have to do with 2FA? For now I changed the login process so it shows the browser signing in with your credentials. Hopefully the error is more clear.

bobby-palko commented:
> I put the personas in a separate folder 

Ah, I overlooked that! `.gitignore` is weird in how it ignores things. It all depends on how you add things, when you add it to the ignore file, etc. If you write it in the `.gitignore` before adding/commiting it, it won't ever get pushed. There is some trickery to commit it once, remove it from the cache, and other stuff so you can at least get it on github and then ignore it after.

I'll take another quick run at it and see if 2FA is causing the login flow to fail; I imagine that might be the case

bobby-palko commented:
lgtm! 👍🏻 things worked now that the browser shows, so it's probably a 2fa issue when running headless
